SOURCE/FD_Jacobian.c

```diff
@@ -287,11 +276,26 @@ void ViscosityDerivatives( grid *mesh, mat_prop *materials, params *model, Npara
                               mesh->bet_n, mesh->div_u,
                               mesh->strain_n, mesh->dil_n, mesh->fric_n, mesh->C_n,
                               model, materials, scaling,
-                              mesh->detadp_n, mesh->ddivpdp_n, mesh->drhodp_n, mesh->eta_n, mesh->rho_n );
+                              mesh->detadp_n, mesh->ddivpdp_n, mesh->drhodp_n, mesh->eta_n, mesh->rho_n, mesh->phase_visc_n );
             // if ( mesh->drhodp_n[c0]<0.0 ) {
             //     printf("%2.2e\n", mesh->drhodp_n[c0]);
             //     exit(1);
             // }
+
+            //----------------------------------------------------------------------------------------------------------------------------------------------------//
+            if ( model->eta_avg==1 ) { // Harmonic
+                mesh->detadexx_n[c0] *= pow(mesh->eta_n[c0] , 2);
```

Comment:
If you write it like this it will add a bit more code, but it also will be clear that you are multiplying and re-writing what is stored there 
`mesh->detadexx_n[c0] = mesh->detadexx_n[c0] * pow(mesh->eta_n[c0] , 2);`

Reply:
I prefer like to use *= it is also very common in many other languages